vars/unitTest.groovy

```diff
@@ -0,0 +1,104 @@
+// vars/functionalTest.groovy
+
+  /**
+   * functionalTest step method
+   *
+   * @param config Map of parameters passed
+   *
+   * config['context']      Context name for SCM to identify the specific
+   *                        stage to update status for.
+   *                        Default is 'test/' + env.STAGE_NAME.
+   *
+   *  Important:
+   *     The SCM status checking for passing may expect a specific name.
+   *
+   *     Matrix stages must override this setting to include matrix axes
+   *     names to ensure a unique name is generated.
+   *
+   *     Or the default name has to be changed in a way that is compatible
+   *     with a future Matrix implementation.
+   *
+   * config['daos_pkg_version']  Version of DAOS package.  Required.
+   *
+   * config['description']       Description to report for SCM status.
+   *                             Default env.STAGE_NAME.
+   *
+   * config['failure_artifacts'] Failure aritfifacts to return.
+   *                             Default env.STAGE_NAME.
+   *
+   * config['ignore_failure']    Ignore test failures.  Default false.
+   * config['inst_repos']        Additional repositories to use.  Optional.
+   *
+   * config['inst_rpms']         Additional rpms to install.  Optional
+   *
+   * config['junit_files']       Junit files to return.
+   *                             Default: 'test_results/*.xml'
+   *
+   * config['NODELIST']          NODELIST of nodes to run tests on.
+   *                             Default env.NODELIST
+   *
+   * config['node_count']        Count of nodes that will actually be used
+   *                             the test.  Default will be based on the
+   *                             enviroment variables for the stage.
+   *
+   * config['stashes']           List of stashes to use.  Default will be
+   *                             baed on the environment variables for the
+   *                             stage.
+   *
+   * config['target']            Target distribution, such as 'centos7',
+   *                             'leap15'.  Default based on parsing
+   *                             environment variables for the stage.
+   *
+   * config['test_rpms']         Set to true to test RPMs being built.
+   *                             Default env.TEST_RPMS.
+   *
+   * config['test_script']       Script to run.
+   *                             Default is 'ci/run_test_main.sh' with
```

Comment:
I noticed prior use of `ci/functional/`.  Should this be ci/run_test/main.sh` for consistency?  I do prefer to have each set of scripts in their own subdir.